qiskit_optimization/algorithms/grover_optimizer.py

```diff
@@ -278,28 +292,30 @@ def _measure(self, circuit: QuantumCircuit) -> str:
         probs = self._get_probs(circuit)
         freq = sorted(probs.items(), key=lambda x: x[1], reverse=True)
         # Pick a random outcome.
-        freq[-1] = (freq[-1][0], 1.0 - sum(x[1] for x in freq[0:len(freq) - 1]))
         idx = algorithm_globals.random.choice(len(freq), 1, p=[x[1] for x in freq])[0]
         logger.info('Frequencies: %s', freq)
-
         return freq[idx][0]
 
     def _get_probs(self, qc: QuantumCircuit) -> Dict[str, float]:
         """Gets probabilities from a given backend."""
         # Execute job and filter results.
         result = self.quantum_instance.execute(qc)
         if self.quantum_instance.is_statevector:
-            state = np.round(result.get_statevector(qc), 5)
+            state = result.get_statevector(qc)
             keys = [bin(i)[2::].rjust(int(np.log2(len(state))), '0')[::-1]
                     for i in range(0, len(state))]
-            probs = [np.round(abs(a) * abs(a), 5) for a in state]
+            probs = [abs(a) * abs(a) for a in state]
+            total = math.fsum(probs)
+            probs = [p / total for p in probs]
             hist = dict(zip(keys, probs))
+            self._circuit_results = state
         else:
             state = result.get_counts(qc)
             shots = self.quantum_instance.run_config.shots
             hist = {}
             for key in state:
                 hist[key[::-1]] = state[key] / shots
+            self._circuit_results = {b[::-1]: (v / shots) ** 0.5 for (b, v) in state.items()}
         hist = dict(filter(lambda p: p[1] > 0, hist.items()))
```

Comment:
You can eliminate this line by removing items with value 0 when `hist` is generated appropriately.